pkg/imgutil/snapshotter.go

```diff
@@ -117,3 +120,71 @@ func stargzExtraLabels(f func(images.Handler) images.Handler, rFlags types.Remot
 func sociExtraLabels(f func(images.Handler) images.Handler, rFlags types.RemoteSnapshotterFlags) func(images.Handler) images.Handler {
 	return socisource.AppendDefaultLabelsHandlerWrapper(rFlags.SociIndexDigest, f)
 }
+
+func SnapshotServiceWithCache(nativeSnapshotter snapshots.Snapshotter) snapshots.Snapshotter {
+	return &CachingSnapshotter{
+		containerdSnapshotter: nativeSnapshotter,
+		statCache:             map[string]snapshots.Info{},
+		usageCache:            map[string]snapshots.Usage{},
+	}
+}
+
+type CachingSnapshotter struct {
+	containerdSnapshotter snapshots.Snapshotter
+	statCache             map[string]snapshots.Info
+	usageCache            map[string]snapshots.Usage
+}
+
+func (snap *CachingSnapshotter) Stat(ctx context.Context, key string) (snapshots.Info, error) {
+	if stat, ok := snap.statCache[key]; ok {
```

Comment:
`snapshots.Info.Updated` might be updated. I think this can't be cached.

Reply:
Hey @ktock ! Thanks for looking at this.

The snapshotter here is NOT meant to do caching across successive calls.

This is meant solely to be used when calling things `image list` and walking through the resulting collection of images, and NEVER for two successive, separate in time, calls.

Furthermore, there is no persistent caching mechanism - same reason.

Reply:
Thanks for clarification. At least it should be documented that this should be used only for short term usage.

But why does it wrap the entire snapshotter API whereas it only patches Stat and Usage? I think this doesn't need to be a wrapper for the entire snapshotter and can be implemented at the caller.

Reply:
You welcome! Yes, I will document.

As for the other question: it does implement the Snapshotter interface so it can easily be swapped in / out in place of the containerd snapshotter.
Of course we can just implement a smaller struct, but that would change the signature.

I am fine with either approach.

Reply:
>  Of course we can just implement a smaller struct, but that would change the signature.

SGTM